source/model.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Created by Ksenia Annenko on 24.02.2017.
+ */
+
+/********************** Model **********************/
+"use strict";
+function Field() {
+    let self = this;
+    let cellsCount = 9;
+    this.fieldCells = [];
+
+    this.updateCell = function (cellNum, symbol) {
+        return self.fieldCells[cellNum] != null ? false : self.fieldCells[cellNum] = symbol;
+    };
+
+    this.getEmptyCells = function () {
+        let emptyCells = [];
+        for (let i = 0; i < cellsCount; i++) {
+            if (self.fieldCells[i] == null) {
+                emptyCells.push(i);
+            }
+        }
+        return emptyCells;
+    };
+
+    this.areCellsAvailable = function () {
+        return self.getEmptyCells().length > 0;
+    }
+}
+
+function WinnerChecker(field) {
+    let winCombinations = [
+        [0, 1, 2], [3, 4, 5], [6, 7, 8],
+        [0, 3, 6], [1, 4, 7], [2, 5, 8],
+        [0, 4, 8], [2, 4, 6]
+    ];
+    let currentWinCombination = null;
+
+    this.getWinner = function (cellNum, symbol) {
+        let winArray = [];
+        for (let i = 0; i < winCombinations.length; i++) {
+            if (winCombinations[i].indexOf(cellNum) != -1) {
+                winArray.push(winCombinations[i]);
+            }
+        }
+        for (let winSegment in winArray) {
+            let isWinnerFound = true;
+            for (let i = 0; i < winArray[winSegment].length; i++) {
+                if (field.fieldCells[winArray[winSegment][i]] != symbol) {
+                    isWinnerFound = false;
+                    break;
+                }
+            }
+            if (isWinnerFound) {
+                return currentWinCombination = winArray[winSegment];
+            }
+        }
+        return false;
+    }
+}
+
+function Player(name, symbol, field) {
+    this.playerName = name;
+    this.playerSymbol = symbol;
+    this.gameField = field;
+    this.myGame = null;
```

Comment:
Yes, we can with gameField, but not with myGame, because it is used in ComputerPlayer and in Game (in playerCreator).